source/MaterialXGenMdl/Nodes/SourceCodeNodeMdl.h

```diff
@@ -0,0 +1,29 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_SOURCECODEMDL_H
+#define MATERIALX_SOURCECODEMDL_H
+
+#include <MaterialXGenShader/Nodes/SourceCodeNode.h>
+
+namespace MaterialX
+{
+
+/// Node implementation using data-driven static source code.
+/// This is the default implementation used for all nodes that 
+/// do not have a custom ShaderNodeImpl class.
+class SourceCodeNodeMdl : public SourceCodeNode
+{
+public:
+    static ShaderNodeImplPtr create();
+
+    void initialize(const InterfaceElement& element, GenContext& context) override;
+    void emitFunctionDefinition(const ShaderNode& node, GenContext& context, ShaderStage& stage) const override;
```

Comment:
As mentioned above this override should probably be removed so we still can emit function definitions into the shader if needed.